app/assets/javascripts/oxalis/model/binary/mappings.js

```diff
@@ -4,29 +4,33 @@
  */
 
 import _ from "lodash";
+import * as THREE from "three";
 import Store from "oxalis/store";
 import Request from "libs/request";
 import ErrorHandling from "libs/error_handling";
 import { doWithToken } from "admin/admin_rest_api";
+import { setMappingAction, setMappingEnabledAction } from "oxalis/model/actions/settings_actions";
+import { createUpdatableTexture } from "oxalis/geometries/materials/abstract_plane_material_factory";
+import UpdatableTexture from "libs/UpdatableTexture";
+import { getRenderer } from "oxalis/controller/renderer";
+import { listenToStoreProperty } from "oxalis/model/helpers/listener_helpers";
+import messages from "messages";
 import type Layer from "oxalis/model/binary/layers/layer";
-import type { DataStoreInfoType } from "oxalis/store";
+import type { DataStoreInfoType, MappingType } from "oxalis/store";
 import type { APIMappingType } from "admin/api_flow_types";
 
-export type MappingType = { [key: number]: number };
+export const MAPPING_TEXTURE_WIDTH = 4096;
 
-// TODO: Non-reactive
 class Mappings {
   mappings: {
     [key: string]: APIMappingType,
-  } = {};
+  };
```

Comment:
Do you think storing `mappings` in the store would be overkill?

Reply:
I don't think that would be overkill, ideally I'd like to make the mappings module entirely stateless, but the textures need to be saved somewhere...

Reply:
Yeah, I think, it's fine to leave the textures out of the store, as it's an implementation detail of the rendering pipeline. The mapping objects itself, however, might fit into the store. Then, this module would only be responsible for the data textures. Fetching and storing the mappings would be done in a saga / in the store. However, I'm fine with doing that at some other point (the benefit is not as high as the other items we are planning to do).

Reply:
For documentation purposes, we agreed on letting the browser handle the caching of the mappings.  I removed the mappings property from mappings.js and changed the code respectively.